stories/2022-03-27_Glass_ML/ML_20th_Century.ipynb

```diff
@@ -0,0 +1,2194 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #5.        from matplotlib.ticker import MultipleLocator
</pre><p>Jennifer Ding:</p><ul><li>Small comment, and <a href="https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcrangelsmith&amp;data=05%7C01%7C%7C4b85c45ca1ca47c7664408da753c5082%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637951199152866481%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&amp;sdata=B8iKuzNKfsw0seRa34rBm35%2BMCfexkFDvYnPQ%2BFDXAM%3D&amp;reserved=0" target="_blank">@crangelsmith</a> should check me on this - but I believe it's better practice to not have imports embedded within functions (e.g. cell 7).</li></ul><p>Joe Cerniglia:</p><p>I would like some of the imports to be embedded in functions, but only in limited cases. The reason the import is embedded is to ensure that the function will run for people who simply want to copy and paste the function into their own code. See additional remark below.</p><p>We could make a note of this “quirkiness” in the text of the story. In my opinion, it is more readable and does not add to the run time. My %timeit test showed the embedded import takes a few nanoseconds more than putting it at the top.</p><p>In [<strong>2</strong>]: %timeit -n 1000 <strong>from</strong> <strong>matplotlib.ticker</strong> <strong>import</strong> MultipleLocator</p><p>812 ns ± 83.2 ns per loop (mean ± std. dev. of 7 runs, 1,000 loops each)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/alan-turing-institute/TuringDataStories/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
<p>Note added September 18, 2022: I will remove the import from inside the function call. Logically, on reflection, I can see it doesn't really make any sense to have it called multiple times.  I will move the import to the location just before the function definition. Thanks.</p>